deep_water_level/src/deep_water_level/train.py

```diff
@@ -37,10 +37,13 @@ def create_model(train_water_line, use_pretrained=False, **kwargs):
         model.load_state_dict(state_dict)
         # Replace the last fully connected layer
         model.fc = nn.Linear(model.fc.in_features, 1)
-    elif train_water_line:
+    elif model_name == "BasicCnnRegressionWaterLine":
         model = BasicCnnRegressionWaterLine(**kwargs)
-    else:
+    elif model_name == "BasicCnnRegression":
         model = BasicCnnRegression(**kwargs)
+    else:
+        raise ValueError(f"Unknown model name: {model_name}")
+
```

Comment:
If we could return `model_name` as part of `kwargs` here, then the model name would get saved with the model as part of model_args and it wouldn't be necessary to pass it as an additional argument in inference scripts!

Reply:
Sounds good, I'll do it in another PR